contracts/VotingFactory.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.16;
+
+import {Create2} from "@openzeppelin/contracts/utils/Create2.sol";
+
+import {ERC1967Proxy} from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+import {IVoting} from "./interfaces/IVoting.sol";
+import {IVotingFactory} from "./interfaces/IVotingFactory.sol";
+
+import {VotingRegistry} from "./VotingRegistry.sol";
+
+/**
+ * @title VotingFactory contract
+ */
+contract VotingFactory is IVotingFactory, Initializable {
+    VotingRegistry public votingRegistry;
+
+    modifier onlyExistingVotingType(string memory votingType_) {
+        _requireExistingVotingType(votingType_);
+        _;
+    }
+
+    /**
+     * @notice The function to initialize the VotingFactory.
+     * @param votingRegistry_ The address of the VotingRegistry contract.
+     *
+     * It binds the VotingRegistry contract to the VotingFactory.
+     */
+    function __VotingFactory_init(address votingRegistry_) external initializer {
+        votingRegistry = VotingRegistry(votingRegistry_);
+    }
+
+    /**
+     * @inheritdoc IVotingFactory
+     */
+    function createVoting(
+        string memory votingType_,
+        IVoting.VotingParams calldata votingParams_
+    ) external onlyExistingVotingType(votingType_) {
+        address voting_ = _deploy(
+            votingType_,
+            abi.encodeWithSelector(IVoting.__Voting_init.selector, votingParams_)
+        );
+
+        _register(votingType_, msg.sender, voting_);
+
+        emit VotingCreated(votingType_, msg.sender, voting_);
+    }
+
+    /**
+     * @inheritdoc IVotingFactory
+     */
+    function createVoting(
```

Comment:
Let's use different (not overloaded) naming

Reply:
Done, changed names to avoid collisions